### PR TITLE
fixes #112 fixes #120 Automate hw-js release process in Travis

### LINKED ISSUE #112: Automate hw-js release process in Travis
**Feature description**

Automate release process steps in Travis script

**Describe the solution you'd like**
Add `deploy` sections in `.travis.yml` script

**Describe alternatives you've considered**
Manual release is not an option

**Additional context**
see skycoin/hardware-wallet#183 as a reference

**Possible implementation**

- [x] Add `make release` target
- [x] Add deploy step for Travis jobs
- [x] Invoke `make release` when Travis build has been triggered by a git tag.
- [x] Release matrix build for ~`osx` and~ `linux`
- [x] Ensure README is in sync with release script
- [x] Explain use of `beta` version
- [x] Deploy to Github releases
- [ ] Pre-release testing steps
- [x] Deploy to `npm`

### LINKED ISSUE #120: Handle EntropyRequest out of band in hw-js library
**Describe the bug**
Handle out-of-band entropy requests.

**Environment information:**

- OS: all
- Platform: discovered on `Linux name 4.19.0-1-amd64 1 SMP Debian 4.19.12-1 (2018-12-22) x86_64 GNU/Linux`
- Firmware Version: skycoin/hardware-wallet@453e780 

**Steps to Reproduce**

TBD , but should happen if Skywallet blocks interaction for more than 2 min  e.g. waiting for button press event.

**Actual behavior**
This is a sample error message displayed while working on #110 

```
Skiping button confirmation! MessageType_EntropyRequest
(node:45924) UnhandledPromiseRejectionWarning: Error: decodeFailureAndPinCode failed
    at devButtonRequestCallback (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:1139:22)
    at devButtonRequestCallback (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:714:5)
    at wordAckLoop (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:1134:13)
    at wordAckCallback (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:1117:5)
    at deviceHandle.read (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:1108:13)
    at BufferReceiver.receiveBuffer (/PATH/TO/skycoin/hardware-wallet-js/buffer-receiver.js:50:7)
    at devHandleCallback (/PATH/TO/skycoin/hardware-wallet-js/device-wallet.js:169:26)
(node:45924) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:45924) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Expected behavior**
`EntropyRequest` should be handled silently by sending `EntropyAck` back to the device.

**Additional context**
See skycoin/hardware-wallet-go#127 , a similar issue for hw-go library

**Possible implementation**
TBD

### PR DESCRIPTION
**Pull request by stdevStark**
Fixes #112 
Fixes #120 

Changes:
- Rules for `npm` package
- Pre release testing procedure
- Out-of-band `EntropyRequest` messages do not interfere with ongoing operation e.g. seed recovery ...

Does this change need to mentioned in CHANGELOG.md?
yes

Requires changes in protobuff specs?
related to PR skycoin/hardware-wallet-protob#52 for issue skycoin/hardware-wallet-protob#51
related to PR skycoin/hardware-wallet-protob#54 for issue skycoin/hardware-wallet-protob#53

Requires testing
yes

Runs in Travis

### CODE REVIEW

**Comment by stdevStark** [comment]
@olemis can not run `travis` command in Docker dev image
created skycoin/skycoin#2321 .

**Comment by olemis** [comment]
> @olemis can not run `travis` command in Docker dev image
> created [skycoin/skycoin#2321](https://github.com/skycoin/skycoin/issues/2321) .

what for ?

**Comment by stdevStark** [comment]
[`Travis npm deploy`](https://docs.travis-ci.com/user/deployment/npm/)

**Comment by olemis** [comment]
> Needs instructions for dealing with the `beta` segment of release versions.

@stdevStark in README , I mean

**Comment by stdevStark** [comment]
> Needs instructions for dealing with the `beta` segment of release versions.

@olemis done

File: ci-scripts/pre-release.checksig.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+dw = require("../skywallet");
+
+dw.setDeviceType(dw.DeviceTypeEnum.USB);
+
+const address = process.argv[2];
+const message = process.argv[3];
+const sig     = process.argv[4];
+
+dw.devCheckMessageSignature(address, message, sig).
```
Comment (olemis) [inline ci-scripts/pre-release.checksig.js]: Wrong . Needs inner hash .

**Comment by olemis** [comment]
merge conflixts

File: README.md [hunk 2]
```diff
@@ -8,4 +8,198 @@ Generate messages.js library:
...
+```sh
+skycoin-cli lisAddresses $WALLET2.wlt
+```
+- Transfer funds to `WALLET2` (if not already done) and check balance
+```sh
+skycoin-cli addressBalance $ADDRESS3
+```
+- Create two transactions. The first one, (i.e. `TXN2`) grabbing funds from first address in `WALLET1` previously recovered in Skywallet device (i.e. `ADDRESS1`). The second one (i.e. `TXN3`) transferring funds from first address in `WALLET2` (i.e. `ADDRESS3`). Receiver in both cases should be second address of `WALLET1` (i.e. `ADDRESS2`) , and change should be sent back to `ADDRESS1`
+```sh
+export TXN2_RAW="$(skycoin-cli createRawTransaction -a $ADDRESS1 -f $WALLET1.wlt -c $ADDRESS1 $ADDRESS2 $AMOUNT)"
```
Comment (olemis) [inline README.md]: No . Go through REST API and create a single transaction spending coins from two addresses. One should be managed by SkyWallet whereas the other should not .

**Pull request merged**